common/src/chain/pos.rs

```diff
@@ -129,6 +129,18 @@ pub fn create_unittest_pos_config() -> PoSChainConfig {
     }
 }
 
+pub fn create_regtest_pos_config() -> PoSChainConfig {
```

Comment:
Ok stupid questions time becuase now I'm confused. Why do we need both unittest and regtest configs? Do we suggest functional tests == regression tests? And how it then relates to `ChainType`?

Reply:
I've been treating the Python tests as the "regression tests".
 
The reason for the added `create_regtest_pos_config()` function is because this would have been a good place to set `difficulty_change_limit` to zero from the timeout PR. And as it was needed anyway, I put it in this PR.

As for the difference in `ChainType`, as far as I can tell where it's needed the most, is in `node-lib/src/runner.rs` which parses the command line options differently than the other chain types. This lets us have different `ChainConfig` settings that can be set from Python but shouldn't be set from the other chain types.

Reply:
> Do we suggest functional tests == regression tests? 

Yes, RegTest is meant to be functional tests. But there has been lots of abuse for all kinds of chain-types. It's not a big deal as it can be fixed later.